src/App/Services/MobileStorageService.cs

```diff
@@ -44,29 +40,42 @@ public class MobileStorageService : IStorageService, IDisposable
 
         public async Task<T> GetAsync<T>(string key)
         {
-            if (_preferenceStorageKeys.Contains(key))
+            if (IsLiteDbKey(key))
             {
-                return await _preferencesStorageService.GetAsync<T>(key);
+                return await _liteDbStorageService.GetAsync<T>(key);
             }
-            return await _liteDbStorageService.GetAsync<T>(key);
+
+            var result = await _preferencesStorageService.GetAsync<T>(key);
+            if (result == null)
+            {
+                // If result is expected in prefs but found in LiteDB, migrate it to prefs
+                result = await _liteDbStorageService.GetAsync<T>(key);
+                if (result != null)
+                {
+                    await _preferencesStorageService.SaveAsync(key, result);
+                    await _liteDbStorageService.RemoveAsync(key);
+                }
+            }
+
+            return result;
```

Comment:
🎨 Even though this is totally fine I like it more when the migration stuff is not done in the same place and we can avoid the comment as well given that the method name is self-explanatory, i.e.:
```c#
public async Task<T> GetAsync<T>(string key)
{
    // ...
    return await _preferencesStorageService.GetAsync<T>(key) ?? TryMigrateLiteDBToPrefsAsync(key);
}

private async Task TryMigrateLiteDBToPrefsAsync(string key)
{
    var currentValue = await _liteDbStorageService.GetAsync<T>(key);
    if (currentValue != null)
    {
        await _preferencesStorageService.SaveAsync(key, currentValue);
        await _liteDbStorageService.RemoveAsync(key);
    }
    return currentValue;
}
```
Also it's easier to read on future versions where the path of migration is less likely to happen.
Thoughts?